wordpress-mobile/WordPress-Android
# Jetpack Section: Improve server creds clarity

Task #14666

This PR improves Server Credentials message clarity with an additional plus icon, interactive highlighted (`colorPrimary`) text followed by a normal text on the Scan and Threat Details screen. 

To test:

1. Install the Jetpack app.
2. Login with an account having a site w/o server credentials e.g. [pressable-jetpack-daily-scan-without-server-creds](http://pressable-jetpack-daily-scan-without-server-creds.mystagingwebsite.com/).
3. Switch to the above site from the site picker and go to the Scan screen from the My Site tab.
4. Notice that "Enter your server credentials to fix threat." is present with an interactive part and a plus icon below the "Fix threat" button.
5. Click on the plus button or the green-colored highlighted text. 
6. Notice that a page is displayed to enter server creds (might need to login).
7. Return to the Scan screen.
8. Click the threat in the threat list on the Scan screen.
9. Notice that the "Threat Details" screen is displayed with the same message below the "Fix threat" button and clicking it opens the same page as on step 6.

~~Note 1: Might need to toggle language to English (United States) as translations already exist for other languages.~~
~~Note 2: Similar string replacements were done earlier here: #13154 and more recently in #14090. I only replaced string values without changing their keys, they were automatically detected and sent to the translators for new translations.~~

a866fcd renames the strings to fix the lint.

Scan screen | Threat Details screen | Scan screen (Dark)  | Threat Details screen (Dark)
-------------|----------------------|---------------------|-----------------------------
![jp_enter_server_creds_scan](https://user-images.githubusercontent.com/1405144/119653190-ab8b0380-be44-11eb-8bf2-635ccb7d8424.png)| ![jp_enter_server_creds_threat](https://user-images.githubusercontent.com/1405144/119653217-b180e480-be44-11eb-9ed1-30a20385e1d6.png)| ![jp_enter_server_creds_scan_dark](https://user-images.githubusercontent.com/1405144/119653233-b5ad0200-be44-11eb-9953-dcdd80280279.png)| ![jp_enter_server_creds_threat_dark](https://user-images.githubusercontent.com/1405144/119653247-b9d91f80-be44-11eb-93e7-dea736aa8787.png)

[Repeat above steps for the WordPress app, notice the same changes are applied to the server creds message (with primary color for WordPress app).]

Scan screen | Threat Details screen | Scan screen (Dark)  | Threat Details screen (Dark)
-------------|----------------------|---------------------|-----------------------------
![wp_enter_server_creds_Scan](https://user-images.githubusercontent.com/1405144/119653654-35d36780-be45-11eb-9ef7-da50d51e99cd.png) | ![wp_enter_server_creds_Threat](https://user-images.githubusercontent.com/1405144/119653666-3a981b80-be45-11eb-8d30-5b6282016370.png) | ![wp_enter_server_creds_Scan_dark](https://user-images.githubusercontent.com/1405144/119653687-41269300-be45-11eb-815b-0a6ff6c9d861.png) | ![wp_enter_server_creds_Threat_dark](https://user-images.githubusercontent.com/1405144/119653709-4552b080-be45-11eb-9fee-5e1348c94e15.png)

**Merge Instructions:**

- Wait for design review
- Remove not ready for merge label
- Merge the PR


## Regression Notes
1. Potential unintended areas of impact
WordPress app

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually WordPress app for the "Enter your server credentials to fix threat." message.

3. What automated tests I added (or what prevented me from doing so)
Updated existing tests for the server credentials message and added tests for the plus icon.

PR submission checklist:
